subsys/net/lib/sockets/sockets_misc.c

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright (c) 2019 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <net/socket.h>
+
+int zsock_gethostname(char *buf, size_t len)
+{
+	const char *p = net_hostname_get();
+
+	strncpy(buf, p, len);
```

Comment:
strncpy() will fail to NUL-terminate buf if there's no space in the buffer for it, and gethostname() always writes a NUL-terminated hostname there

Reply:
POSIX is very cunning (as obviously, strncpy is what everyone implementsit with): http://pubs.opengroup.org/onlinepubs/9699919799/functions/gethostname.html

> The returned name shall be null-terminated, except that if namelen is an insufficient length to hold the host name, then the returned name shall be truncated and it is unspecified whether the returned name is null-terminated.

Reply:
I love POSIX. I really do. :)